### PR TITLE
Adopt parameter types; Move to cucumber 3

### PR DESCRIPTION
**Pull request by mwhipple**

### CODE REVIEW

**Comment by mwhipple** [comment]
This involves a fair amount of changes and needs some more cleanup but I'll likely continue to work on that in the background and try to get this released so that it can be used and necessary changes started in any local ruby code.

File: features/argument_transforms/whitespace.feature [hunk 1]
```diff
@@ -10,6 +10,13 @@ Feature: Whitespace
       """
     Then the value of the response body is equal to `<expected>`
 
+  Examples:
+    | input              | expected     |
```
Comment (mwhipple) [inline features/argument_transforms/whitespace.feature]: the content below intentionally includes tabs so this isn't going to line up consistently

**Comment by mwhipple** [comment]
@timmc-bcov If you want to 👍 this I can get it out, the reviews are getting a bit backlogged

**Review by timmc-bcov — APPROVED** [review]
I couldn't give it a thorough review, but nothing jumped out at me.

**Comment by mwhipple** [comment]
Released as 0.12.0. This actually is using a Cucumber 4 RC which I tried using to resolve an arity resolution problem with cucumber-expressions...but I think I'll leave it at that version unless there are issues. At this point I'll also try to focus on getting a v1 out which should include publishing of changelogs, etc.

**Comment by mwhipple** [comment]
That didn't last long...Cucumber 4 has an issue on OS X so `0.12.1` was released using cucumber 3

**Pull request merged**